internal/git/gitsync_processor.go

```diff
@@ -192,21 +207,57 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1alpha1.GitSync
 	return hash.String(), nil
 }
 
+// ApplyOwnerShipReference adds ownerships to resources for GitSync
+func ApplyOwnerShipReference(manifest string, gitSync *v1alpha1.GitSync) ([]byte, error) {
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	decoded, _, err := decode([]byte(manifest), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	obj, ok := decoded.(metav1.Object)
+	if !ok {
+		return nil, errors.New("decoded manifest is not a metaV1 object")
+	}
+	ownerRef := metav1.OwnerReference{
+		APIVersion:         gitSync.APIVersion,
+		Kind:               gitSync.Kind,
+		Name:               gitSync.Name,
+		UID:                gitSync.UID,
+		Controller:         ptr.To(true),
+		BlockOwnerDeletion: ptr.To(true),
+	}
+	obj.SetOwnerReferences([]metav1.OwnerReference{ownerRef})
```

Comment:
I think we should probably append this onto any existing OwnerReferences just in case rather than replacing the entire array